server/src/test/java/keywhiz/service/resources/automation/v2/SecretResourceConfigNewSecretOwnershipStrategyInferTest.java

```diff
@@ -0,0 +1,94 @@
+package keywhiz.service.resources.automation.v2;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.dropwizard.jackson.Jackson;
+import java.io.IOException;
+import java.util.Base64;
+import java.util.UUID;
+import keywhiz.IntegrationTestRule;
+import keywhiz.KeywhizService;
+import keywhiz.TestClients;
+import keywhiz.api.automation.v2.CreateGroupRequestV2;
+import keywhiz.api.automation.v2.CreateSecretRequestV2;
+import keywhiz.api.automation.v2.SecretDetailResponseV2;
+import okhttp3.OkHttpClient;
+import okhttp3.Response;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.RuleChain;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class SecretResourceConfigNewSecretOwnershipStrategyInferTest {
+  private static final ObjectMapper mapper = KeywhizService.customizeObjectMapper(Jackson.newObjectMapper());
+  private static final Base64.Encoder encoder = Base64.getEncoder();
+  private static final Integer RESPONSE_CODE_CREATED = 201;
+  private static final String AUTOMATION_CLIENT_GROUP_NAME = "Security";
+
+  private SecretResourceTest secretResourceTest;
+
+  OkHttpClient mutualSslClient;
+
+  @ClassRule public static final RuleChain chain = IntegrationTestRule.rule("keywhiz-test-newSecretOwnershipStrategyInfer.yaml");
+
+  @Before public void setUp() {
+    mutualSslClient = TestClients.mutualSslClient();
+    secretResourceTest = new SecretResourceTest();
+    secretResourceTest.mutualSslClient = mutualSslClient;
+  }
```

Comment:
Creating an instance of a test inside another test is definitely thinking outside the box, but that's a pattern that's usually avoided. If there are related tests that share helper methods/resources then typically you can either have a common abstract base class that they inherit from (say, AbstractSecretResourceTestBase, and IIRC that class can have @ Before etc. methods to properly initialize its own fields, although I'd have to double-check the JUnit docs on that). The other one would be to move the shared utility method to a dedicated class (SecretResourceTestHelper?) and have them both use that instead.

Reply:
Ah ok I will look around other files and see how I can refactor the tests. I am fairly certain that creating a new instance of another test class is a common thing among several other tests .